### ISSUE TITLE
Warning: Redis::expire(): Unknown expiration modifier ''

### ISSUE THREAD

**Issue by dmeziere** [phpredis/phpredis#2385]
### Expected behaviour

When an EXPIRE call is issued without expiration modifier (3rd parameter), the call should only provide 2 parameters :

`EXPIRE [key] [ttl]`

### Actual behaviour

When an EXPIRE call is issued without expiration modifier, a null value is added as third parameter :

`EXPIRE [key] [ttl] <null>`

When using Symfony and snc/redis-bundle, this results in an exception :

`Warning: Redis::expire(): Unknown expiration modifier ''`

Even at the lower layer, using telnet, it results in an error from Redis :

`-ERR wrong number of arguments for 'expire' command`

### I'm seeing this behaviour on
- OS: Ubuntu 20.04.6 LTS
- Redis: 5.0.7
- PHP: 8.0.30
- phpredis: 6.0.0

**Comment by yatsukhnenko** [comment]
The modifier parameter is optional, you can check this with `php --re redis`
```
        Method [ <internal:redis> public method expire ] {

          - Parameters [3] {
            Parameter #0 [ <required> string $key ]
            Parameter #1 [ <required> int $timeout ]
            Parameter #2 [ <optional> ?string $mode = null ]
          }
          - Return [ Redis|bool ]
        }
```

Also simple tests script shows that it works correctly
```php
$redis = new Redis();
$redis->set("foo", "bar", 10);
var_dump($redis->ttl("foo"));
var_dump($redis->expire("foo", 42));
var_dump($redis->ttl("foo"));
```
Output
```
int(10)
bool(true)
int(42)
```

If you have any problem with Symfony and snc/redis-bundle it may be better to open issue in their bug tracker

**Comment by aglowienka** [comment]
I guess It is not related to the bundle. I have the same issue, starting from `v6.0.0`.

My guess is that when `null` value (default) is passed as 3rd argument (`mode`), it is casted to an empty string (looking at `redis_commands.c` line 6005 and 6008 this seems to be the case).

**Comment by kosolapovvp** [comment]
I have the same issue.

I change example by adding NULL to the expire command
```php
$redis = new Redis();
$redis->set("foo", "bar", 10);
var_dump($redis->ttl("foo"));
var_dump($redis->expire("foo", 42, null));
var_dump($redis->ttl("foo"));
```
Output
```
int(10)
PHP Deprecated:  Redis::expire(): Passing null to parameter #3 ($mode) of type ?string is deprecated in /var/www/test.php on line 6

Deprecated: Redis::expire(): Passing null to parameter #3 ($mode) of type ?string is deprecated in /var/www/test.php on line 6
PHP Warning:  Redis::expire(): Unknown expiration modifier '' in /var/www/test.php on line 6

Warning: Redis::expire(): Unknown expiration modifier '' in /var/www/test.php on line 6
bool(false)
int(10)
```
I see deprecation about `Passing null to parameter #3 ($mode) of type ?string is deprecated` and expired method was failed

**Comment by dmeziere** [comment]
@yatsukhnenko It depends on the versions of software you have :

    $ php -a
    Interactive mode enabled

    php > $redis = new Redis();
    php > $redis->set("foo", "bar", 10);
    php > var_dump($redis->ttl("foo"));
    int(6)
    php > var_dump($redis->expire("foo", 42));
    bool(false)
    php > var_dump($redis->ttl("foo"));
    int(-2)

**Comment by kosolapovvp** [comment]
@dmeziere Maybe you just use a short TTL in your last example
```
php -a
Interactive shell

php > $redis = new Redis();
php > $redis->set("foo", "bar", 100);
php > var_dump($redis->ttl("foo"));
int(95)
php > var_dump($redis->expire("foo", 42));
bool(true)
php > var_dump($redis->ttl("foo"));
int(36)
```

**Comment by johanwilfer** [comment]
I see the same with Symfony framework on 6.0.0, reverting to 5.3.7 resolves the issue. Helpful discussion here: https://github.com/phpredis/phpredis/issues/2383

And as I understand it this relates to the proxy classes:
https://github.com/phpredis/phpredis/issues/2383#issuecomment-1717911716
https://github.com/phpredis/phpredis/issues/2383#issuecomment-1718954879
> Those proxy classes are generated from the reflection that extension provides. If they are generated wrong, it means extension needs to fix information they provide via reflection.

We have installed the previous versions and held packages to not have this roll out in production, as I have not yet figured out how to not use the proxy classes and everything explodes on v6.

**Comment by ostrolucky** [comment]
I think issue is this 
```
php > var_dump((new ReflectionClass('Redis'))->getMethod('expire')->getParameters()[2]->getDefaultValue());
php shell code:1:
NULL
```

phpredis maintainers: If you don't want folks to pass `null` or empty string  there, set default value to something valid please.

**Comment by yatsukhnenko** [comment]
https://github.com/phpredis/phpredis/pull/2388

**Comment by ahmed-bhs** [comment]
@yatsukhnenko @johanwilfer any idea how to force apt install php8.2-redis to install phpredis 5.3.7 extension instead of 6.0.0 ?

**Comment by michael-grunder** [comment]
I'd need to know what distro you are using to check whether you would be able to specify an older version.

I don't see an older package in Ubuntu or debian.

Another route is to install via pecl:

To install PECL:

```bash
pecl install php-pear
```

Install PhpRedis 5.3.7

```bash
pecl install redis-5.3.7
```

If that doesn't work you can just download 5.3.7 from PECL or here and build from source

**Comment by aglowienka** [comment]
He's most likely using Ondrej Sury's PPA for Ubuntu (or his repo for Debian). It's impossible to downgrade there, because there's only one (the latest) version available.

**Comment by johanwilfer** [comment]
I use Ondrej Sury's PPA and what I did was to download the previous version from the PPA manually, specifically this: https://launchpad.net/~ondrej/+archive/ubuntu/php/+files/php8.2-redis_5.3.7++-1+ubuntu20.04.1+deb.sury.org+1_amd64.deb

And then install with dpkg -i in our CI / local env. If you have a server not yet upgrade you can use apt-hold to avoid upgrading.

But I am probably switching to pecl that can install a specific version instead so rollback is easier in cases like this.

**Comment by fliespl** [comment]
I think this was fixed in 6.0.1 on pecl.
